### ISSUE TITLE
NUnit reports PASSED for `Values`-driven parametrized tests with no cases

### ISSUE THREAD

**Issue by toolmagic** [nunit/nunit#3349]
NUnit 3.12.0
NUnit.ConsoleRunner 3.10.0 
.NET Framework 4.7.2 class library project

I added obviously incorrect test definitions and run with NUnit Console. I expected they failed but NUnit reported PASSED! 
Expected behavior: tests are ignored or failed.

```csharp
    public class Class1
    {
        [Test]
        public void Test1([Values] object value) // Expected failed or ignored but PASSED reported!
        {
            Assert.Fail();
        }

        [Test]
        [Combinatorial]
        public void Test2(ConsoleColor color) // Expected failed or ignored but PASSED reported!
        {
            Assert.Fail();
        }

        [Test]
        [Sequential]
        public void Test3(ConsoleColor color) // Expected failed or ignored but PASSED reported!
        {
            Assert.Fail();
        }
    }
```

**Comment by ChrisMaddock** [comment]
I agree this is a framework issue, rather than an engine/console one, if any change is needed. Moving over now.

**Comment by stevenaw** [comment]
@nunit/framework-team I'm adding this to the 4.6 milestone as it pairs well with https://github.com/nunit/nunit/issues/3663 and will help ensure we have the desirable behaviour for both.

Based on the discussion in https://github.com/nunit/nunit/issues/3663#issuecomment-3549189063 it sounds like we may prefer the default here to be `Failed` instead of `Passed` or `inconslusive`

**Comment by OsirisTerje** [comment]
Could be, but in this case the test never reach the body of the test.  It seems to fail (or pass as is the case for this one) before it hits the test itself.  So, this could be an Inconclusive too.   But Fail could also work, since it obviously will not do any good, and should be reacted upon by the developer. 

With the adapter, when I have been in doubt, I have added a config parameter that sets the default but can be changed to something else if the user prefer.  We could do the same, both for this and for the #3663 issue.  Just a thought......

**Comment by stevenaw** [comment]
I had forgotten your original nuance in that comment about `Inconclusive`. It is possible that default has changed in 4.5. I'll have to revisit this when I'm able. I had purposely excluded testing this when looking at 3663 as I expected this to also be changing within the same release. Perhaps now there may not be anything to do here anymore. 

I agree with the configuration-based option for `NoTests` as well. I had not realized you had already added some adapter support. Perhaps it would only be a small enhancement to add on after 3663.

**Comment by stevenaw** [comment]
This is still an issue as near as I can tell in version 4.5. I have not yet tested in against a pre-release that includes the fix for https://github.com/nunit/nunit/issues/3663

Currently in 4.5 something like this: 
```
[Test]
        public void Test1([Values] object value) // Expected failed or ignored but PASSED reported!
        {
            Assert.Fail();
        }
```

Yields the following during discovery:
```
<test-suite type='ParameterizedMethod' id='0-1001' name='Test1' fullname='Issue3349.Tests.Test1' classname='Issue3349.Tests' runstate='Runnable' testcasecount='0' />
```

However it is not actually runnable of course. I'll revisit this again once we can get another pre-release out https://github.com/nunit/nunit/issues/3663

**Comment by OsirisTerje** [comment]
@stevenaw I have no issue with using Failed now, just wanted to mention Inconclusive.  I did NOT want to stop you !

**Comment by stevenaw** [comment]
Not at all! I always appreciate design agreement _before_ starting work, which you provided 🙂

My window to investigate has closed for this afternoon here but I'll be continuing as time allows later. I aim to be able to close this in some manner, even if as "fixed elsewhere", as part of 4.6-related work

**Comment by stevenaw** [comment]
Having looked into this, I can confirm this to be a framework issue which is no longer reproducible. It appears to have been fixed sometime between 3.13 and 3.14.

**Comment by OsirisTerje** [comment]
I added fixed_in_newer_version to indicate it is actually fixed, and not abandoned.

**Comment by stevenaw** [comment]
Thanks! I had been looking for that
